packages/xo-web/src/xo-app/hub/resource.js

```diff
@@ -0,0 +1,255 @@
+              })
+            )
+          )
+          success(_('hubImportNotificationTitle'), _('successfulInstall'))
+        } catch (_error) {
+          error(_('hubImportNotificationTitle'), _error.message)
+        }
+        markHubResourceAsInstalled(id)
+      },
+      async create() {
+        const { isPoolCreated, installedTemplates } = this.state
+        const { name } = this.props
+        if (getXoaPlan(process.env.XOA_PLAN) === 'Community') {
+          subscribeAlert()
+          return
+        }
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm poolPredicate={isPoolCreated} {...props} />
+          ),
+          header: (
+            <span>
+              <Icon icon='add-vm' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+        const { $pool } = resourceParams.pool
+        const template = find(installedTemplates, { $pool })
+        if (template !== undefined) {
+          this.props.router.push(
+            `/vms/new?pool=${$pool}&template=${template.id}`
+          )
+        } else {
+          throw new Error(`can't find template for pool: ${$pool}`)
+        }
+      },
+      async deleteTemplates(__, { name }) {
+        const { isPoolCreated } = this.state
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm
+              delete
+              multi
+              poolPredicate={isPoolCreated}
+              {...props}
+            />
+          ),
+          header: (
+            <span>
+              <Icon icon='vm-delete' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+        const _templates = filter(this.state.installedTemplates, template =>
+          find(resourceParams.pools, { $pool: template.$pool })
+        )
+        await deleteTemplates(_templates)
+      },
+      updateSelectedInstallPools(_, selectedInstallPools) {
+        return {
+          selectedInstallPools,
+        }
+      },
+      updateSelectedCreatePool(_, selectedCreatePool) {
+        return {
+          selectedCreatePool,
+        }
+      },
+      redirectToTaskPage() {
+        this.props.router.push('/tasks')
+      },
+    },
+    computed: {
+      installedTemplates: (_, { id, templates }) =>
+        filter(templates, ['other.xo:resource:xva:id', id]),
+      isTemplateInstalledOnAllPools: ({ installedTemplates }, { pools }) =>
+        installedTemplates.length > 0 &&
+        pools.every(
+          pool =>
+            installedTemplates.find(template => template.$pool === pool.id) !==
+            undefined
+        ),
+      isTemplateInstalled: ({ installedTemplates }) => pool =>
+        installedTemplates.find(template => template.$pool === pool.id) ===
+        undefined,
+      isPoolCreated: ({ installedTemplates }) => pool =>
+        installedTemplates.find(template => template.$pool === pool.id) !==
+        undefined,
```

Comment:
I don't understand the names of these computeds, are you sure they are correct?

Reply:
I renamed'em by the request of reviewers, that was not my choice.

Reply:
It appears incorrect to me, can you point me to the relevant comment so I can see the reasoning please?

Reply:
https://github.com/vatesfr/xen-orchestra/pull/4442#discussion_r327535425